### ISSUE TITLE
Sveltekit DEV server doesn't error when importing js code that uses client-side apis

### ISSUE THREAD

**Issue by Fractal-Tess** [sveltejs/kit#9264]
### Describe the bug

Just as the title suggests, the sveltekit development server SSRs components that import javascript code that uses the document object.  

(Rant: I'm not sure if this is a feature or anything, but having to debug errors that only present themselves during the deployment of production code is headache-inducing at best.)

### Reproduction

Here is a reproduction repo - https://github.com/Fractal-Tess/sveltekit-ssrbug

Steps:
```bash
git clone https://github.com/Fractal-Tess/sveltekit-ssrbug 
cd sveltekit-ssrbug 
pnpm i
pnpm dev
```
Go ahead and look at `localhost:5173/` to see that the component has rendered fine (both on the server and the client)
Now build the app and run it.
```
pnpm build
node build
```
Go ahead and navigate back to `localhost:5173/` again, but this time there is SSR rendering exception. 

### Logs

```Shell
ReferenceError: document is not defined
    at +page.svelte:6:14
    at Object.$$render (/node_modules/.pnpm/svelte@3.55.1/node_modules/svelte/internal/index.mjs:1892:22)
    at Object.default (root.svelte:42:40)
    at eval (/node_modules/.pnpm/@sveltejs+kit@1.9.2_svelte@3.55.1+vite@4.1.4/node_modules/@sveltejs/kit/src/runtime/components/layout.svelte:8:41)
    at Object.$$render (/node_modules/.pnpm/svelte@3.55.1/node_modules/svelte/internal/index.mjs:1892:22)
    at root.svelte:41:39
    at $$render (/node_modules/.pnpm/svelte@3.55.1/node_modules/svelte/internal/index.mjs:1892:22)
    at Object.render (/node_modules/.pnpm/svelte@3.55.1/node_modules/svelte/internal/index.mjs:1900:26)
    at Module.render_response (/node_modules/.pnpm/@sveltejs+kit@1.9.2_svelte@3.55.1+vite@4.1.4/node_modules/@sveltejs/kit/src/runtime/server/page/render.js:180:29)
    at async Module.render_page (/node_modules/.pnpm/@sveltejs+kit@1.9.2_svelte@3.55.1+vite@4.1.4/node_modules/@sveltejs/kit/src/runtime/server/page/index.js:310:10)
```
```


### System Info

```Shell
System:
    OS: Linux 6.1 Arch Linux
    CPU: (16) x64 AMD Ryzen 7 2700 Eight-Core Processor
    Memory: 6.86 GB / 15.55 GB
    Container: Yes
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.14.2 - ~/.nvm/versions/node/v18.14.2/bin/node
    Yarn: 1.22.19 - ~/.local/share/pnpm/yarn
    npm: 9.3.0 - ~/.local/share/pnpm/npm
  Browsers:
    Firefox: 110.0
  npmPackages:
    @sveltejs/adapter-node: 1.2.0 => 1.2.0 
    @sveltejs/kit: ^1.5.0 => 1.9.2 
    svelte: ^3.54.0 => 3.55.1 
    vite: ^4.0.0 => 4.1.4 ```
```

### Severity

annoyance

**Comment by chvanam** [comment]
I haven't tried to repro but maybe this can help in your case?

https://www.youtube.com/watch?v=l1AB5f0Xyd4

**Comment by Rich-Harris** [comment]
Fixed by #9276, though you can also work around it by externalising the `vidstack` dependency — just move it out of `devDependencies` and into `dependencies`, and `adapter-node` won't bundle it

**Comment by tymon3568** [comment]
> Fixed by #9276, though you can also work around it by externalising the `vidstack` dependency — just move it out of `devDependencies` and into `dependencies`, and `adapter-node` won't bundle it

i use videstack too, but when not use (page.ts)
```
export const ssr = false
export const prerender = false
```
The player broken ui when first load, and i don't know how to fix it, vidstack already in dependencies
